config/confignet/confignet.go

```diff
@@ -5,10 +5,53 @@ package confignet // import "go.opentelemetry.io/collector/config/confignet"
 
 import (
 	"context"
+	"fmt"
 	"net"
 	"time"
 )
 
+// TransportType represents a type of network transport protocol
+type TransportType string
+
+const (
+	TCP        TransportType = "tcp"
+	TCP4       TransportType = "tcp4"
+	TCP6       TransportType = "tcp6"
+	UDP        TransportType = "udp"
+	UDP4       TransportType = "udp4"
+	UDP6       TransportType = "udp6"
+	IP         TransportType = "ip"
+	IP4        TransportType = "ip4"
+	IP6        TransportType = "ip6"
+	Unix       TransportType = "unix"
+	Unixgram   TransportType = "unixgram"
+	UnixPacket TransportType = "unixpacket"
```

Comment:
The constants should start with the type as we do in other components e.g. `TransportTypeUnixPacket`. It might be worth putting such guidelines in docs.

Reply:
Totally agree.  I can document all the enum naming decisions once we decide something on the `Type` or not `Type` discussion.